### PR TITLE
SDK: make the dataset cache directory customizable

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This is useful for people whose home directory is too small/not fast enough. It also lets us make the tests less hacky.

### How has this been tested?

Unit tests. I also ran my old demos to make sure the default cache location remained the same.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- ~~[ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly~~
- [x] I have added tests to cover my changes
- ~~[ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat-sdk/cvat_sdk/core/client.py [hunk 1]
```diff
@@ -43,6 +47,9 @@ class Config:
     verify_ssl: Optional[bool] = None
     """Whether to verify host SSL certificate or not"""
 
+    cache_dir: Path = attrs.field(converter=Path, default=_DEFAULT_CACHE_DIR)
```
Comment (zhiltsov-max) [inline cvat-sdk/cvat_sdk/core/client.py:50]: I'm not sure if the empty value should be allowed now. Probably, it's better to disallow this until the relying code supports it.

Comment (SpecLad) [reply cvat-sdk/cvat_sdk/core/client.py:50]: Empty value? What do you mean?

Comment (zhiltsov-max) [reply cvat-sdk/cvat_sdk/core/client.py:50]: I mean, the user can change the default value and pass the empty string in their code during `Client` configuration. Will this work?

Comment (SpecLad) [reply cvat-sdk/cvat_sdk/core/client.py:50]: `Path('')` is the current directory, so that will set the cache path to the current directory.

**Review by zhiltsov-max — APPROVED** [review]

**Review by nmanovic — APPROVED** [review]

**Pull request merged**